src/utils/dnd.ts

```diff
@@ -22,3 +23,98 @@ export const updateLabels = (
 
   return updatedLabels;
 };
+
+type UpdatedLabels = {
+  labels: Label[];
+  answeredLabels: AnsweredLabel[];
+};
+
+const findLabelIndex = (labels: AnsweredLabel[], id: string): number =>
+  labels?.findIndex(({ expected }) => expected.id === id) ?? -1;
+
+type MoveIndices = {
+  srcIdx: number;
+  distIdx: number;
+};
+
+type MoveContext = {
+  labels: Label[];
+  answeredLabels: AnsweredLabel[];
+  srcDroppableId: string;
+  distDroppableId: string;
+  srcLabelIndex: number;
+};
+
+export const trackLabelsChanges = ({
+  srcDroppableId,
+  distDroppableId,
+  labels,
+  answeredLabels,
+  srcLabelIndex,
+}: MoveContext): UpdatedLabels => {
+  const handleMoveFromAll = ({
```

Comment:
Maybe I'd finally put these three functions outside the trackLabelsChanges to improve a bit the readability of this function 😅